packages/canvas/canvas-mesh/src/CanvasMeshRenderer.ts

```diff
@@ -148,34 +148,40 @@ export class CanvasMeshRenderer
 
         if (canvasPadding > 0)
         {
-            const paddingX = canvasPadding / Math.abs(mesh.worldTransform.a);
-            const paddingY = canvasPadding / Math.abs(mesh.worldTransform.d);
+            const { a, b, c, d } = mesh.worldTransform;
+
             const centerX = (x0 + x1 + x2) / 3;
             const centerY = (y0 + y1 + y2) / 3;
 
             let normX = x0 - centerX;
             let normY = y0 - centerY;
 
-            let dist = Math.sqrt((normX * normX) + (normY * normY));
-
-            x0 = centerX + ((normX / dist) * (dist + paddingX));
-            y0 = centerY + ((normY / dist) * (dist + paddingY));
+            let dx = (a * normX) + (c * normY);
```

Comment:
Please add an explanation of the math/reasoning. For example:

https://github.com/pixijs/pixi.js/blob/cc5f3a1a1662ef69188e199839df57af5cdf9879/packages/core/src/filters/FilterSystem.ts#L600

Reply:
I added some comments and renamed some of the variables to make it easier to follow.